src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
```

Comment:
[CODE PERFORMANCE] The `die` method uses `filter` to remove the dead animal from the `Animal.alive` array, which is good practice as it avoids nesting loops. However, calling this method directly from `setHealth` could lead to performance issues if `setHealth` is called multiple times in a loop, as it would repeatedly create new arrays. Consider checking the health of the animal before calling `filter` to ensure that it is necessary.